homeassistant/components/reolink/button.py

```diff
@@ -0,0 +1,133 @@
+"""Component providing support for Reolink button entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import GuardEnum, Host, PtzEnum
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkButtonEntityDescriptionMixin:
+    """Mixin values for Reolink button entities."""
+
+    method: Callable[[Host, int], Any]
+
+
+@dataclass
+class ReolinkButtonEntityDescription(
+    ButtonEntityDescription, ReolinkButtonEntityDescriptionMixin
+):
+    """A class that describes button entities."""
+
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+
+
+BUTTON_ENTITIES = (
+    ReolinkButtonEntityDescription(
+        key="ptz_stop",
+        name="PTZ stop",
+        icon="mdi:pan",
+        supported=lambda api, ch: api.supported(ch, "pan_tilt"),
+        method=lambda api, ch: api.set_ptz_command(ch, command=PtzEnum.stop.value),
+    ),
+    ReolinkButtonEntityDescription(
+        key="ptz_left",
+        name="PTZ left",
+        icon="mdi:pan",
+        supported=lambda api, ch: api.supported(ch, "pan_tilt"),
+        method=lambda api, ch: api.set_ptz_command(ch, command=PtzEnum.left.value),
+    ),
+    ReolinkButtonEntityDescription(
+        key="ptz_right",
+        name="PTZ right",
+        icon="mdi:pan",
+        supported=lambda api, ch: api.supported(ch, "pan_tilt"),
+        method=lambda api, ch: api.set_ptz_command(ch, command=PtzEnum.right.value),
+    ),
+    ReolinkButtonEntityDescription(
+        key="ptz_up",
+        name="PTZ up",
+        icon="mdi:pan",
+        supported=lambda api, ch: api.supported(ch, "pan_tilt"),
+        method=lambda api, ch: api.set_ptz_command(ch, command=PtzEnum.up.value),
+    ),
+    ReolinkButtonEntityDescription(
+        key="ptz_down",
+        name="PTZ down",
+        icon="mdi:pan",
+        supported=lambda api, ch: api.supported(ch, "pan_tilt"),
+        method=lambda api, ch: api.set_ptz_command(ch, command=PtzEnum.down.value),
+    ),
+    ReolinkButtonEntityDescription(
```

Comment:
Should this have a config entity category?

Reply:
Yes, I have added the EntityCategory.CONFIG